AndroidSDKLocation/src/main/java/com/leanplum/LocationManagerImplementation.java

```diff
@@ -301,21 +301,17 @@ private void updateTrackedGeofences() {
           trackedGeofenceIds.add(geofenceId);
           //TODO: stateBeforeBackground doesn't get persisted.
           // If the app goes to the background and terminates, stateBeforeBackground will be reset.
-          if (isInBackground && !Util.isInBackground() && stateBeforeBackground != null
-              // This is triggered only for in-app messages, since backgroundGeofences are only for
-              // pushes.
-              // TODO(aleks): This would not work for in-app messages if we have the same geolocation
-              // triggering it, as a locally triggered push notification.
-              && !backgroundGeofences.contains(geofence)) {
+          if (isInBackground && !Util.isInBackground() && stateBeforeBackground != null) {
             Number lastStatus = (Number) stateBeforeBackground.get(geofenceId);
             Number currentStatus = (Number) lastKnownState.get(geofenceId);
             if (currentStatus != null && lastStatus != null) {
+              // Only foreground geofences should be triggered here
               if (GeofenceStatus.shouldTriggerEnteredGeofence(lastStatus, currentStatus)) {
-                maybePerformActions(geofence, "enterRegion");
+                maybePerformActions(geofence, "enterRegiron", true);
```

Comment:
whoops. Error in the string - let's fix that.
Also, is there any test we can write to ensure that this kind of thing gets caught automatically ?